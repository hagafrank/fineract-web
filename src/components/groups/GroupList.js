import React from "react";
import PropTypes from "prop-types";
import { FormattedMessage } from "react-intl";

const GroupList = ({ groups }) => (
  <table className="table">
    <thead>
      <tr>
        <th>
          <FormattedMessage id="GroupList.name" />
        </th>
        <th>
          <FormattedMessage id="GroupList.accountNumber" />
        </th>
        <th>
          <FormattedMessage id="GroupList.externalId" />
        </th>
        <th>
          <FormattedMessage id="GroupList.status" />
        </th>
        <th>
          <FormattedMessage id="GroupList.office" />
        </th>
      </tr>
    </thead>
    <tbody>
      {groups.map((group) => {
        return (
          <tr key={group.id}>
            <td>{group.name}</td>
            <td>{group.accountNo}</td>
            <td>{group.externalId}</td>
            <td>{group.status.value}</td>
            <td>{group.officeName}</td>
          </tr>
        );
      })}
    </tbody>
  </table>
);

GroupList.propTypes = {
  groups: PropTypes.array.isRequired
};

export default GroupList;
