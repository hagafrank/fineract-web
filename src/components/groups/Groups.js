import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import GroupList from "./GroupList";
import Spinner from "../common/Spinner";
import { FormattedMessage } from "react-intl";
import { loadGroups } from "../../redux/actions/groupActions";

class Groups extends React.Component {
  constructor() {
    super();

    this.state = {
      loading: false
    };
  }

  toggleSpinner = (show) => {
    this.setState({ loading: show });
  };

  componentDidMount() {
    this.toggleSpinner(true);
    this.props
      .loadGroups()
      .then(() => this.toggleSpinner(false))
      .catch((error) => {
        this.toggleSpinner(false);
        console.error("Loading groups failed. " + error);
      });
  }

  render() {
    return (
      <>
        <h2>
          <FormattedMessage id="Groups.header" />
        </h2>
        {this.state.loading ? (
          <Spinner />
        ) : (
          <GroupList groups={this.props.groups} />
        )}
      </>
    );
  }
}

Groups.propTypes = {
  groups: PropTypes.array.isRequired,
  loadGroups: PropTypes.func.isRequired
};

function mapStateToProps(state) {
  return {
    groups: state.groups.groups || []
  };
}

const mapDispatchToProps = {
  loadGroups
};

export default connect(mapStateToProps, mapDispatchToProps)(Groups);
