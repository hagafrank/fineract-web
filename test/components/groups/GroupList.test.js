import React from "react";
import { groups } from "../../mock_data/groups/groups";
import GroupList from "../../../src/components/groups/GroupList";
import createComponentWithIntl from "../../helpers/jest-snapshot/createComponentWithIntl";

test("groups list should be rendered", () => {
  const component = createComponentWithIntl(<GroupList groups={groups} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
