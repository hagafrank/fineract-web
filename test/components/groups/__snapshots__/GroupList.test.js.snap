// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`groups list should be rendered 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th>
        Name
      </th>
      <th>
        Account #
      </th>
      <th>
        External Id
      </th>
      <th>
        Status
      </th>
      <th>
        Office
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Clientes    Plazo Fijo
      </td>
      <td>
        000000001
      </td>
      <td />
      <td>
        Active
      </td>
      <td>
        Big Data Bank
      </td>
    </tr>
    <tr>
      <td>
        SUPER MERCHANT
      </td>
      <td>
        000000002
      </td>
      <td />
      <td>
        Active
      </td>
      <td>
        ARCA NETWORK
      </td>
    </tr>
    <tr>
      <td>
        tryphena women group
      </td>
      <td>
        000000003
      </td>
      <td />
      <td>
        Active
      </td>
      <td>
        Head Office
      </td>
    </tr>
    <tr>
      <td>
        Fiter
      </td>
      <td>
        000000005
      </td>
      <td />
      <td>
        Active
      </td>
      <td>
        Head Office
      </td>
    </tr>
  </tbody>
</table>
`;
